pyne/r2s.py

```diff
@@ -231,30 +231,42 @@ def total_photon_source_intensity(m, tag_name, sub_voxel=False):
             intensity += vol * np.sum(sv_data)
     return intensity
  
-def tag_e_bounds(m, e_bounds, tag_name='e_bounds'):
-    """This function tags the energy boundaries to the PyMOAB mesh
+def tag_phtn_src_info(m, value, tag_name, dtype=float):
+    """This function tags the information of photon source to the PyMOAB mesh
     instance as a sparse tag for the purpose of photon source sampling.
 
     Parameters
     ----------
     m : PyNE Mesh
        The mesh-based photon emission density distribution in p/cm3/s.
-    e_bounds : numpy array or list
-        The energy boundaries of a photon source.
+    value : numpy array, list or float
+        The information to be tagged, could be:
+        - e_bounds: The energy boundaries of a photon source, numpy array or
+            list, unit: eV.
+        - decay_time: The decay time after shutdown, unit: s.
+        - total_photon_source_intensity: The total photon source intensity at
+            current decay time, unit: p/s.
+        - version: The PyNE version that changes the information or
```

Comment:
I think it would be more clear to create separate functions for each of these purposes:

  - `tag_e_bounds`
  - `tag_source_intensity`
  - `tag_decay_time`
 - `tag_version`

This will make it easier to check that the incoming data for the tag is valid (iterable of floats for e_bounds, single float for decay time, etc.) in those higher level functions before calling the more general function. If there's a decent amount of common code between those three functions, then we could probably separate it out into a more generic function for tagging data call it where appropriate within `tag_e_bounds`, `tag_source_intensity`, `tag_decay_time`, and `tag_version` respectively.

Reply:
From the look of your nicely written `tag_phtn_src_info` function, these will essentially just be wrapper functions, but I think they provide some clarity and separation of concerns about the data types and requirements for these different pieces of information in the photon source file.